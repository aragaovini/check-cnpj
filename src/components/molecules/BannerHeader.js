import React from 'react';
import styled from 'styled-components';
import TitleDescription from '../atoms/TitleDescription';
import MaterialIcon from 'material-icons-react';

const BackgroundContainer = styled.div`
    display: flex;
    flex: 1;
    align-items: center;
    padding: 12px;
    background: rgba(165,54,255,1);
    background: -moz-linear-gradient(top, rgba(165,54,255,1) 0%, rgba(116,5,216,1) 90%, rgba(110,0,212,1) 100%);
    background: -webkit-gradient(left top, left bottom, color-stop(0%, rgba(165,54,255,1)), color-stop(90%, rgba(116,5,216,1)), color-stop(100%, rgba(110,0,212,1)));
`;

const AvatarImage = styled.img`
    width: 64px;
`;

const IconContainer = styled.div`
    margin: 18px;
`;

const FlexBlock = () => {
    return <span style={{flex: 1}}/>;
}

const BannerHeader = (props) => {
    const { title, code, image } = props;
    return (
        <BackgroundContainer className="banner-header">
            <IconContainer>
                <MaterialIcon color="#fff" size="40" icon="trending_up" />
            </IconContainer>
            <div>
                <TitleDescription className="title-container" title={title} />
                <div className="code-container">{code}</div>
            </div>
            <FlexBlock />
            <AvatarImage src={image} alt="avatar-profile" />
        </BackgroundContainer>
    );
}

export default BannerHeader;