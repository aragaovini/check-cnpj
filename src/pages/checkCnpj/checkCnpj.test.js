import React from 'react';
import { shallow } from 'enzyme';
import CheckCnpj from './';
import BannerHeader from '../../components/molecules/BannerHeader';

const Page = <CheckCnpj />;

describe('Check-Cnpj Page', () => {

    it('should render without crashing', () => {
        const component = shallow(Page);
        expect(component).toMatchSnapshot();
    });

    it('should render only one banner', () => {
        const wrapper = shallow(Page);
        const banner = wrapper.find(BannerHeader);
        expect(banner.length).toBe(1);
    });

    it('should state cnpj be right', () => {
        const wrapper = shallow(Page);
        const input = wrapper.find('.cnpj-input');
        input.simulate('change', {target: {value: '43.454.362/0001-65'}});
        expect(wrapper.state().cnpj).toBe('43.454.362/0001-65');
    });

    it('should state be empty', () => {
        const wrapper = shallow(Page);
        const input = wrapper.find('.cnpj-input');
        input.simulate('change', {target: {value: ''}});
        expect(wrapper.state().cnpj).toBe('');
    });
});