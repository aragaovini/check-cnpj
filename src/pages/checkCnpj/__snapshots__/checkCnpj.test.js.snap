// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check-Cnpj Page should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckCnpj />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BannerHeader
          code="#0980"
          image="avatar.svg"
          title="Nova cotação"
        />,
        <styled.h3>
          Buscar por CNPJ ou empresa
        </styled.h3>,
        <div
          className="container-input"
        >
          <InputElement
            className="cnpj-input"
            mask="99.999.999/9999-99"
            maskChar=" "
            onChange={[Function]}
            value=""
          />
        </div>,
        <Button
          clickMethod={[Function]}
          text="Ok"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "code": "#0980",
          "image": "avatar.svg",
          "title": "Nova cotação",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Buscar por CNPJ ou empresa",
        },
        "ref": null,
        "rendered": "Buscar por CNPJ ou empresa",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <InputElement
            className="cnpj-input"
            mask="99.999.999/9999-99"
            maskChar=" "
            onChange={[Function]}
            value=""
          />,
          "className": "container-input",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "cnpj-input",
            "mask": "99.999.999/9999-99",
            "maskChar": " ",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "clickMethod": [Function],
          "text": "Ok",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BannerHeader
            code="#0980"
            image="avatar.svg"
            title="Nova cotação"
          />,
          <styled.h3>
            Buscar por CNPJ ou empresa
          </styled.h3>,
          <div
            className="container-input"
          >
            <InputElement
              className="cnpj-input"
              mask="99.999.999/9999-99"
              maskChar=" "
              onChange={[Function]}
              value=""
            />
          </div>,
          <Button
            clickMethod={[Function]}
            text="Ok"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "code": "#0980",
            "image": "avatar.svg",
            "title": "Nova cotação",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Buscar por CNPJ ou empresa",
          },
          "ref": null,
          "rendered": "Buscar por CNPJ ou empresa",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <InputElement
              className="cnpj-input"
              mask="99.999.999/9999-99"
              maskChar=" "
              onChange={[Function]}
              value=""
            />,
            "className": "container-input",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "cnpj-input",
              "mask": "99.999.999/9999-99",
              "maskChar": " ",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "clickMethod": [Function],
            "text": "Ok",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
